modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/js/components/FileEntryPicker.tsx

```diff
@@ -0,0 +1,304 @@
+
+		selectFileButtonRef.current?.focus();
+	};
+
+	const validateFrontendTokenDefinitionFile = async (file: File) => {
+		post(objectToFormData({file}), '/validate-file')
+			.then(async (response) => {
+				const data = await response.json();
+
+				if (response.ok) {
+					setFeedback({
+						displayType: 'success',
+						message: data.message,
+					});
+				}
+				else {
+					throw data.message;
+				}
+			})
+			.catch((error) => {
+				disableFormSubmitButton(true, portletNamespace);
+
+				setDangerFeedback(
+					error
+						? error
+						: Liferay.Language.get(
+								'your-upload-failed-to-complete-please-try-again-or-contact-support-if-the-error-persists'
+						  )
+				);
+			});
+	};
+
+	const announceJSONValidation = () => {
+		setIsValidatingJSON(true);
+
+		if (!alertRole) {
+			setAlertRole('alert');
+		}
+
+		setInfoFeedback(
+			Liferay.Language.get(
+				'the-frontend-token-definition-json-file-is-being-uploaded-and-validated'
+			)
+		);
+	};
+
+	const handleFileInputChange = async ({
+		target,
+	}: ChangeEvent<HTMLInputElement>) => {
+		const filePath = target.value;
+
+		if (!filePath.endsWith('.json')) {
+			setDangerFeedback(
+				Liferay.Language.get(
+					'the-format-is-not-valid-please-upload-a-valid-frontend-token-definition-json-file'
+				)
+			);
+
+			disableFormSubmitButton(true, portletNamespace);
+
+			return;
+		}
+
+		announceJSONValidation();
+
+		if (target.files === null) {
+			setDangerFeedback(
+				Liferay.Language.get(
+					'your-upload-failed-to-complete-please-try-again-or-contact-support-if-the-error-persists'
+				)
+			);
+
+			return;
+		}
+
+		const file = target.files[0];
+
+		await validateFrontendTokenDefinitionFile(file);
+
+		setFrontendTokenDefinition(await readInputFile(file));
+
+		setIsValidatingJSON(false);
+	};
+
+	useEffect(() => {
+		if (frontendTokenDefinition && externalReferenceCode) {
+			post(
+				objectToFormData({
+					companyId: Liferay.ThemeDisplay.getCompanyId(),
+					externalReferenceCode,
+				}),
+				'/tokens-info'
+			)
+				.then((response) => response.json())
+				.then((data) => {
+					if (data.message) {
+						setFeedback({
+							displayType: 'success',
+							message: data.message,
+						});
+					}
+				});
+		}
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	const fileInputId = `${portletNamespace}file`;
+	const frontendTokenDefinitionDescriptionId = `${portletNamespace}frontendTokenDefinitionDescription`;
+
+	return (
+		<LearnResourcesContext.Provider value={learnResources}>
+			<label
+				aria-describedby={frontendTokenDefinitionDescriptionId}
+				className="d-block"
+				htmlFor={fileInputId}
+				tabIndex={0}
+			>
+				{Liferay.Language.get('frontend-token-definition-json-upload')}
+			</label>
+
+			<Text
+				as="p"
+				color="secondary"
+				id={frontendTokenDefinitionDescriptionId}
+				size={3}
+			>
+				{Liferay.Language.get(
+					'the-frontend-token-definition-is-a-json'
+				)}{' '}
+
```

Comment:
Do you need that blank space? If so, you can use string literal here to avoid these issues, `${Liferay.Language.get('the-frontend-token-definition-is-a-json')} ` otherwise in theory just remove the space in white would do the trick.

Reply:
Applied the changes you comment on the old and new PR, thanks @matuzalemsteles